lib/dpl/provider/code_deploy.rb

```diff
@@ -53,17 +53,38 @@ def github_revision
       end
 
       def push_app
-        deployment = code_deploy.create_deployment({
+        data = code_deploy.create_deployment({
           revision:               revision,
           application_name:       options[:application]      || option(:application_name),
           deployment_group_name:  options[:deployment_group] || option(:deployment_group_name),
           description:            options[:description]      || default_description
         })
-        log "Triggered deployment #{deployment.deployment_id.inspect}."
+        log "Triggered deployment #{data.deployment_id.inspect}."
+        return unless options[:wait_until_deployed]
+        print "Deploying "
+        deployment = wait_until_deployed(data[:deployment_id])
+        print "\n"
+        if deployment[:status] == 'Succeeded'
+          log "Deployment successful."
+        else
+          error "Deployment failed."
+        end
       rescue Aws::CodeDeploy::Errors::DeploymentLimitExceededException => exception
         error(exception.message)
       end
 
+      def wait_until_deployed(deployment_id)
+        deployment = nil
```

Comment:
You'd probably want

```ruby
deployment = {}
```

which agrees with the general type that `loop` is expected to assign to `deployment`. (I understand that the value is overwritten, but it reads better to me.)